core/Algorytm.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CTP.core
+{
+    internal class Algorytm
+    {
+        public static DataTable VtoMm(double VLow, double VHigh, double MmLow, double MmHigh, DataTable originalData)
+        {
+            //double offset1 = -VLow;
+            //double offset2 = -MmLow;
+            if (VLow ==  VHigh || MmLow == MmHigh) throw new ArgumentException("Division by 0 in VtoMm conversion has ocurred\n");
+
+            DataTable returnTable = new DataTable();
+            returnTable.Columns.Add("Mm", typeof(double));
+
+            double scale = (MmHigh-MmLow)/(VHigh-VLow);
+
+            foreach (DataRow row in originalData.Rows) 
+            {
+
+                double newval = (row.Field<double>(0) - VLow) * scale + MmHigh;
```

Comment:
Widzę, że zakładamy, że tabela, która przychodzi jako piewszą kolumnę ma wartości, natomiast w pliku `measurement.txt` jest ona dopiero w 1 kolumnie.

Reply:
@JN0122, odnosząc się do obu komentarzy - zakładałem, że funkcja dostanie tylko jedne wartości napięcia i przeliczy je na odległość. Jeżeli ma operować na surowym DataTable z parsera, to robi się bałagan, trzeba sprawdzać co jest czasem, co impulsem a co napięciem, dodatkowo dla każdego czujnika są różne skale. Myślę, że taką złożoną funkcjonalność lepiej jest rozbić na mniejsze fragmenty, ale to zależy w jaki sposób chcesz finalnie używać tej funkcji, więc możemy się dogadać.